python/src/uagents/resolver.py

```diff
@@ -293,9 +304,29 @@ async def resolve(self, destination: str) -> Tuple[Optional[str], List[str]]:
                     k=min(self._max_endpoints, len(endpoints)),
                 )
         except Exception as e:
-            print(f"Error in AlmanacApiResolver when resolving {destination}: {e}")
+            LOGGER.error(
+                f"Error in AlmanacApiResolver when resolving {destination}: {e}"
+            )
 
-        return await self._almanac_resolver.resolve(address)
+        return None, []
+
+    async def resolve(self, destination: str) -> Tuple[Optional[str], List[str]]:
+        """
+        Resolve the destination using the Almanac API.
+        If the resolution using API fails, it retries using the Almanac Contract.
+
+        Args:
+            destination (str): The destination address to resolve.
+
+        Returns:
+            Tuple[Optional[str], List[str]]: The address and resolved endpoints.
+        """
+        address, endpoints = await self._api_resolve(destination)
+        return (
+            (address, endpoints)
+            if address is not None
+            else await self._almanac_contract_resolver.resolve(address)
```

Comment:
Should be 
`else await self._almanac_contract_resolver.resolve(destination)`
Because address would be None

Reply:
Good catch! That's a bug